providers/aws-ec2/src/test/java/org/jclouds/aws/ec2/features/AWSSubnetApiMockTest.java

```diff
@@ -82,6 +118,82 @@ public void createSubnetInRegion_options() throws Exception {
       assertPosted(DEFAULT_REGION, "Action=CreateSubnet&VpcId=vpc-1a2b3c4d&CidrBlock=10.0.1.0/24&DryRun=true&AvailabilityZone=us-east-1a");
    }
 
+   public void deleteSubnetInRegion() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueue(DEFAULT_REGION,
+            new MockResponse().setBody("<DeleteSubnetResponse xmlns=\"http://ec2.amazonaws.com/doc/2016-09-15/\">\n" +
+                    "  <requestId>7a62c49f-347e-4fc4-9331-6e8eEXAMPLE</requestId>\n" +
+                    "  <return>true</return>\n" +
+                    "</DeleteSubnetResponse>"));
+
+      subnetApi().deleteSubnetInRegion(DEFAULT_REGION, "subnet-9d4a7b6c");
+      
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DeleteSubnet&SubnetId=subnet-9d4a7b6c");
+   }
+
+   public void describeSubnetInRegion() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueue(DEFAULT_REGION,
+            new MockResponse().setBody(describeSubnetsResponse));
+
+      FluentIterable<Subnet> results = subnetApi().describeSubnetsInRegion(DEFAULT_REGION);
+      Subnet result = Iterables.getOnlyElement(results);
+      assertEquals(result.getSubnetId(), "subnet-9d4a7b6c");
+      assertEquals(result.getSubnetState().value(), "available");
+      assertEquals(result.getVpcId(), "vpc-1a2b3c4d");
+      assertEquals(result.getCidrBlock(), "10.0.1.0/24");
+      assertEquals(result.getAvailabilityZone(), "us-east-1a");
+      assertEquals(result.getAvailableIpAddressCount(), 251);
+      
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DescribeSubnets");
+   }
+
+   public void describeSubnetInRegionWithFilter() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueue(DEFAULT_REGION,
+            new MockResponse().setBody(describeSubnetsResponse));
+
+      FluentIterable<Subnet> results = subnetApi().describeSubnetsInRegionWithFilter(DEFAULT_REGION,
+            ImmutableMultimap.of("vpc-id", "vpc-1a2b3c4d", "availabilityZone", "us-east-1a"));
+      Subnet result = Iterables.getOnlyElement(results);
+      assertEquals(result.getSubnetId(), "subnet-9d4a7b6c");
+      assertEquals(result.getSubnetState().value(), "available");
+      assertEquals(result.getVpcId(), "vpc-1a2b3c4d");
+      assertEquals(result.getCidrBlock(), "10.0.1.0/24");
+      assertEquals(result.getAvailabilityZone(), "us-east-1a");
+      assertEquals(result.getAvailableIpAddressCount(), 251);
+      
+      assertPosted(DEFAULT_REGION, "Action=DescribeRegions");
+      assertPosted(DEFAULT_REGION, "Action=DescribeSubnets&Filter.1.Name=vpc-id&Filter.1.Value.1=vpc-1a2b3c4d&Filter.2.Name=availabilityZone&Filter.2.Value.1=us-east-1a");
+   }
+
+   public void describeSubnetInRegionWhenEmpty() throws Exception {
+      enqueueRegions(DEFAULT_REGION);
+      enqueue(DEFAULT_REGION,
+            new MockResponse().setBody(describeSubnetsEmptyResponse));
```

Comment:
These "empty" tests trigger the same jclouds logic than the other ones. Instead of enqueuing an OK response with an empty list, just enqueue a 404 response without a body to trigger the `@Fallback` defined in the api methods. The expected results should be the same.